pkg/server/auth/server.go

```diff
@@ -186,6 +187,11 @@ func (s *AuthServer) oauth2Config(scopes []string) *oauth2.Config {
 		scopes = append(scopes, scopeGroups)
 	}
 
+	// Ensure "offline_access" scope is always present for refresh tokens.
+	if !contains(scopes, oidc.ScopeOfflineAccess) {
```

Comment:
One thing we could check is, do all providers support offline access?

Maybe we make a quick list of possibles (other than Dex) and check whether or not they allow it.

Some examples, Keycloak (we have customers using it), Google and anybody else you can think of?

What happens if the provider does not allow it?

Reply:
There is a list of all oauth providers who support OIDC: https://en.wikipedia.org/wiki/List_of_OAuth_providers, we can make a shortlist from that list as a start?

Reply:
User can now omit offline_access from scope if they need to